actionpack/lib/action_dispatch/http/parameters.rb

```diff
@@ -54,7 +54,7 @@ def parameters
                  rescue EOFError
                    query_parameters.dup
                  end
-        params.merge!(path_parameters)
+        params.merge!(path_parameters_utf8_encoding)
```

Comment:
`set_binary_encoding` bellow should set all values as UTF8 already, if they don't specify another encode